app/Models/Beatmapset.php

```diff
@@ -524,8 +531,28 @@ public static function searchES(array $params = [])
                 break;
         }
 
+        // recommended difficulty
+        if ($params['recommended'] && $params['user'] !== null) {
+            // TODO: index convert difficulties and handle them.
+            $mode = Beatmap::modeStr($param['mode'] ?? Beatmap::MODES['osu']);
```

Comment:
more like `$params`. Also is it converting back and forth between string and int ಠ_ಠ

Reply:
sure is :trollface: 
Either that or something should be done about `User::statistics`